### PR TITLE
Starting Python3 and Django 1.11 upgrade 

### PR DESCRIPTION
**Pull request by tyrelsouza**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[tox]
+envlist = py27,py36
```
Comment (morganwahl) [inline tox.ini]: Please test on py35 as well, since that's what we'll have on Ubuntu 16.04.

File: dbfilestorage/storage.py [hunk 2]
```diff
@@ -6,7 +7,7 @@
 from django.db.models import Q
 from django.core.files.base import ContentFile
 from django.core.files.storage import Storage
-from django.core.urlresolvers import reverse
+from django.core.urlresolvers import reverse_lazy
```
Comment (morganwahl) [inline dbfilestorage/storage.py]: Why the change to reverse_lazy?

Comment (tyrelsouza) [reply dbfilestorage/storage.py]: ran into this problem http://www.boxtricks.com/keyerror-en-us-when-doing-a-reverse-lookup-in-urls-py/

File: dbfilestorage/models.py [hunk 3]
```diff
@@ -11,8 +11,8 @@ class DBFile(models.Model):
     b64 = models.TextField()
     mtime = models.DateTimeField(auto_now=True)
 
-    def __unicode__(self):
-        return u"{name} <{content_type}>".format(
+    def __str__(self):
+        return "{name} <{content_type}>".format(
```
Comment (morganwahl) [inline dbfilestorage/models.py]: I think this will fail on Python 2 if `self.name` has non-ASCII

Comment (morganwahl) [reply dbfilestorage/models.py]: Apply the `python_2_unicode_compatible` decorator to the class to have `__str__` be used for `__unicode__` on Python 2.

Comment (morganwahl) [reply dbfilestorage/models.py]: Futurize could use some exposition, but here's their example: http://python-future.org/compatible_idioms.html#custom-str-methods

File: tests/tests.py [hunk 4]
```diff
@@ -10,11 +12,14 @@
...
 from django.test.utils import override_settings
 
+from future.utils import python_2_unicode_compatible
+
 
 PROJECT_ROOT = os.path.dirname(os.path.realpath(__file__))
 DEFAULT_FILE_STORAGE = "dbfilestorage.storage.DBFileStorage"
 
 
+@python_2_unicode_compatible
```
Comment (morganwahl) [inline tests/tests.py]: This is not what this is for. See above.

File: dbfilestorage/storage.py [hunk 5]
```diff
@@ -52,8 +53,8 @@ def _save(self, name, content, max_length=None):
                 content.file.seek(0)
                 read_data = content.file.read()
         else:
-            read_data = content.file.encode('utf8')
-        b64 = read_data.encode('base64')
+            read_data = content.file
```
Comment (morganwahl) [inline dbfilestorage/storage.py]: The next line will fail unless `content.file` is bytes (`str` type on python 2) here. Previously it looks like you expected it to be characters, since you encoded it to bytes with UTF-8.

Probably good to add a :param content: to document this.

Comment (morganwahl) [reply dbfilestorage/storage.py]: How is `_save` called? Would be good to check that whatever is calling it with something where `content.file` is characters still works on Python 2.

File: dbfilestorage/urls.py [hunk 6]
```diff
@@ -1,5 +1,5 @@
 from django.conf.urls import url
-import views
+from dbfilestorage import views
```
Comment (morganwahl) [inline dbfilestorage/urls.py]: `from __future__ import absolute_import` would give you the same behavior on 2 and 3.

File: dbfilestorage/storage.py [hunk 7]
```diff
@@ -52,8 +53,8 @@ def _save(self, name, content, max_length=None):
                 content.file.seek(0)
                 read_data = content.file.read()
         else:
```
Comment (morganwahl) [inline dbfilestorage/storage.py]: looks like this is sort of an existing bug: https://docs.djangoproject.com/en/1.11/ref/files/file/#django.core.files.File.file

or at least, a mis-use of the API that isn't guaranteed to always work.

I think the whole `if hasattr .. else..` block could be replaced with

```py
with content.open('rb'):
    read_data = content.read()
```

**Comment by morganwahl** [comment]
I'm tempted to say you should do `from __future__ import absolute_import, print_function, unicode_literals` in every single module, but I don't see any places it would make a difference with the current code.

File: dbfilestorage/storage.py (hunk 5 above)
Comment (tyrelsouza) [reply dbfilestorage/storage.py]: code is tested in python 2 and 3 on CircleCi, so it wont even run the python3 test unless it's still compatible with python3

File: tests/tests.py (hunk 4 above)
Comment (tyrelsouza) [reply tests/tests.py]: Ah so my unicode literals i added before this actually fixed the problem i thought i was running into. gotcha.

File: dbfilestorage/storage.py (hunk 7 above)
Comment (tyrelsouza) [reply dbfilestorage/storage.py]: I'll try that
